### PR TITLE
Add support for \e escape sequences in c# strings and characters

### PR DESCRIPTION
**Pull request by CyrusNajmabadi**
Language feature approved by the LDM for the any-time bucket: https://github.com/dotnet/csharplang/issues/7400

### CODE REVIEW

File: src/Workspaces/SharedUtilitiesAndExtensions/Compiler/CSharp/EmbeddedLanguages/VirtualChars/CSharpVirtualCharService.cs
Comment (CyrusNajmabadi) [inline src/Workspaces/SharedUtilitiesAndExtensions/Compiler/CSharp/EmbeddedLanguages/VirtualChars/CSharpVirtualCharService.cs:361]: will add IDE tests as well.  But just wanted to get the ball rolling on this.

**Comment by jaredpar** [comment]
Don't forget to update language feature status.md

File: src/Compilers/CSharp/Test/Syntax/LexicalAndXml/LexicalTests.cs
Comment (jcouv) [inline src/Compilers/CSharp/Test/Syntax/LexicalAndXml/LexicalTests.cs:845]: `RegularPreview` is the default. I think you intend to test using `RegularNext`, which will be updated mechanically to `Regular13` once it becomes available.

File: src/Compilers/CSharp/Test/Syntax/LexicalAndXml/LexicalErrorTests.cs
Comment (CyrusNajmabadi) [inline src/Compilers/CSharp/Test/Syntax/LexicalAndXml/LexicalErrorTests.cs:325]:
```suggestion
            ParserErrorMessageTests.ParseAndValidate(test, TestOptions.RegularNext);
```

File: src/Compilers/CSharp/Test/Syntax/LexicalAndXml/LexicalErrorTests.cs
Comment (CyrusNajmabadi) [inline src/Compilers/CSharp/Test/Syntax/LexicalAndXml/LexicalErrorTests.cs:341]:
```suggestion
            ParserErrorMessageTests.ParseAndValidate(test, TestOptions.RegularNext);
```

File: src/Compilers/CSharp/Test/Syntax/LexicalAndXml/LexicalErrorTests.cs
Comment (CyrusNajmabadi) [inline src/Compilers/CSharp/Test/Syntax/LexicalAndXml/LexicalErrorTests.cs:357]:
```suggestion
            ParserErrorMessageTests.ParseAndValidate(test, TestOptions.RegularNext);
```

File: src/Compilers/CSharp/Test/Syntax/LexicalAndXml/LexicalTests.cs
Comment (CyrusNajmabadi) [inline src/Compilers/CSharp/Test/Syntax/LexicalAndXml/LexicalTests.cs:845]:
```suggestion
            var token = LexToken(text, TestOptions.RegularNext);
```

File: src/Compilers/CSharp/Test/Syntax/LexicalAndXml/LexicalTests.cs
Comment (CyrusNajmabadi) [inline src/Compilers/CSharp/Test/Syntax/LexicalAndXml/LexicalTests.cs:1024]:
```suggestion
            var token = LexToken(text, TestOptions.RegularNext);
```

File: src/Compilers/CSharp/Test/Syntax/LexicalAndXml/LexicalTests.cs
Comment (CyrusNajmabadi) [reply src/Compilers/CSharp/Test/Syntax/LexicalAndXml/LexicalTests.cs:845]: done.  thanks!

File: docs/Language Feature Status.md
Comment (jcouv) [inline docs/Language Feature Status.md:17]: >[CyrusNajmabadi](https://github.com/CyrusNajmabadi) [](http://example.com/codeflow?start=194&length=51)

Put whoever reviews as reviewers (me and someone else)

**Review by jcouv — APPROVED** [review]
Compiler side LGTM Thanks (iteration 6)

File: docs/Language Feature Status.md
Comment (CyrusNajmabadi) [reply docs/Language Feature Status.md:17]: will do.  when i get a second reviewer :)

Comment (CyrusNajmabadi) [inline docs/Language Feature Status.md:17]:
```suggestion
| [Escape character](https://github.com/dotnet/csharplang/issues/7400) | N/A | [In Progress](https://github.com/dotnet/roslyn/pull/70497) | [CyrusNajmabadi](https://github.com/CyrusNajmabadi) | [jcouv](https://github.com/jcouv) |  | [CyrusNajmabadi](https://github.com/CyrusNajmabadi) |
```

**Comment by CyrusNajmabadi** [comment]
@dotnet/roslyn-compiler @jaredpar  for another set of eyes.

File: docs/Language Feature Status.md
Comment (CyrusNajmabadi) [inline docs/Language Feature Status.md:17]:
```suggestion
| [Escape character](https://github.com/dotnet/csharplang/issues/7400) | N/A | [In Progress](https://github.com/dotnet/roslyn/pull/70497) | [CyrusNajmabadi](https://github.com/CyrusNajmabadi) | [jcouv](https://github.com/jcouv) |  | [CyrusNajmabadi](https://github.com/CyrusNajmabadi) | 
```

**Comment by CyrusNajmabadi** [comment]
@dotnet/roslyn-compiler  @RikkiGibson  can i get a set of eyes on this small change?  thanks!

**Review by RikkiGibson — APPROVED** [review]

File: docs/Language Feature Status.md
Comment (CyrusNajmabadi) [inline docs/Language Feature Status.md:17]:
```suggestion
| [Escape character](https://github.com/dotnet/csharplang/issues/7400) | N/A | [In Progress](https://github.com/dotnet/roslyn/pull/70497) | [CyrusNajmabadi](https://github.com/CyrusNajmabadi) | [jcouv](https://github.com/jcouv), [RikkiGibson](https://github.com/RikkiGibson) |  | [CyrusNajmabadi](https://github.com/CyrusNajmabadi) | 
```

**Pull request merged**
